object-manipulation-2/src/defaults.js

```diff
@@ -1,9 +1,24 @@
 /* exported defaults */
 // create a function to move the properties of the source object into the target object that aren't already in the target object
-// assign the values of target to source to make them equal
-// then assign all the properties in source back to target
+// loop through the source object
+// loop through the target object
+// if the key matches to the key in the target
+// delete the key and continue the function
+// loop through the updated keys object
+// assign each value to the target object
+// return the target object from the function.
 
 function defaults(target, source) {
-  Object.assign(source, target);
-  Object.assign(target, source);
+  for (var key in source) {
+    for (var keyT in target) {
+      if (key === keyT) {
+        delete source[key];
```

Comment:
Oh no, you don't want to delete any properties from `source`, that object should remain unchanged at the end of the function. Objects and arrays passed as arguments should never be changed unless that is the specific operation you are meant to do. This function is meant to make changes to `target`, but `source` is just meant to be examined, not changed.